### DISCUSSION TITLE
Welcome to aria-telegram-mirror-bot Discussions!

### DISCUSSION THREAD

**Discussion by arghyac35** [arghyac35/aria-telegram-mirror-bot#60] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by xavierxross** [comment]
Nice

**Comment by kumarhariomrai** [comment]
cool

**Comment by kumarhariomrai** [comment]
unzipping isn't working in bot

**Reply by arghyac35** [reply]
Give more information like what type of file u tried to unzip and error log.

**Comment by BOI-Hub** [comment]
What changes to  make in docker file.. I changed the repo in it with my username and deployed still it did fetched the changes i made in repo.. Btw i deployed it in heroku.

**Reply by arghyac35** [reply]
You cannot change the repo in dockerfile like that. You need to first build a package and push it to gh container then u can use yours.

Or you can pick this [commit](https://github.com/arghyac35/aria-telegram-mirror-bot/commit/6d9471660724e5462e2fb8b13316ebae12eae086), it will build the package while deploying to heroku. So, it will take changes from your repo.

**Reply by BOI-Hub** [reply]
Ohh, picking commit that i can do...
And thanks for replying

**Comment by vigneshgarrapally** [comment]
Hello, I tried to deploy in Heroku using free dynos. While doing so, I made a small mistake in the constant.js file( wrote True instead of true). After changing also, it is not updating. May I know the exact process to be done after editing the constant.js file

**Reply by arghyac35** [reply]
When u change the gist, the url gets updated, so u need to update the latest URL in action secret.

**Reply by BOI-Hub** [reply]
If u r talking about gist link... U need to update new raw link of that gist after editing to CONSTANT_URL in github secret...
Link gets change everytime after editing...

**Reply by vigneshgarrapally** [reply]
Okay, thanks. I have updated the URL. So does that automatically reflect in Heroku or do we have to do anything in addition?

**Reply by BOI-Hub** [reply]
Go to action then redploy all jobs

**Reply by vigneshgarrapally** [reply]
I am new to all these. Can you please let me know who to do that exactly.
Thanks

**Reply by arghyac35** [reply]
> I am new to all these. Can you please let me know who to do that exactly.
> Thanks

"Exactly" follow step 5 of https://github.com/arghyac35/aria-telegram-mirror-bot/blob/master/heroku_guide.md#deployment

**Reply by vigneshgarrapally** [reply]
Thanks a lot.

**Comment by norm-git** [comment]
Hello
#todo
Can add possibility to recursively search in subfolders. ok no problem for API Drive, because we are using Service accounts.

**Comment by ghost** [comment]
Neither Service Accounts Nor Token Provided. Exiting...

This Error coming

**Reply by arghyac35** [reply]
From now on I won't reply to this type of issue. These issues are lack of following herku_guide and readme.

**Reply by lonexd3vil** [reply]
> From now on I won't reply to this type of issue. These issues are lack of following herku_guide and readme.

dude i followed the guide...i made git secrets and they are working but it cant seem to find me accounts repo..why?

**Comment by anoxdleo** [comment]
what extension to give in HEROKU_API_KEY , HEROKU_MAIL & HEROKU_APP_NAME ? 

I'm getting this error 
![image](https://user-images.githubusercontent.com/28385639/119254052-cace2a80-bbd1-11eb-8007-c9fb68f68ecb.png)

**Comment by lonexd3vil** [comment]
am getting this error, 
![Screenshot (23)](https://user-images.githubusercontent.com/57116517/119344124-e65e3180-bcb4-11eb-9a0f-67a7db76ffb1.png)

**Comment by snehitap** [comment]
Hey Arghya, i m curious about one thing.
I know we don't need to create app first in heroku, workflow does it automatically.. 

But the question is this.. What if I already have app in heroku and then hit deploy button with that same app name in github secret will the workflow deploy in that app.. I am asking this because i have 500gb storage in that app & i don't want to lose that storage..

**Reply by arghyac35** [reply]
> will the workflow deploy in that app

Yes, it will deploy in that, if an app exists then it doesn't create a new app.

**Reply by alokkkk288** [reply]
> Hey Arghya, i m curious about one thing.
> I know we don't need to create app first in heroku, workflow does it automatically..
> 
> But the question is this.. What if I already have app in heroku and then hit deploy button with that same app name in github secret will the workflow deploy in that app.. I am asking this because i have 500gb storage in that app & i don't want to lose that storage..

Can you please tell me how did you get 500 gb storage

**Reply by snehitap** [reply]
Some months back Heroku reduced the storage to 215 Gb for some days, then restored to 500GB to compensate ig..

**Comment by tomyprs** [comment]
Bro i got ETELEGRAM: 404 Not Found how to fix that

**Comment by EdithX** [comment]
Application Logs
2021-06-05T05:47:34.659842+00:00 app[worker.1]:     at Function.Module._load (node:internal/modules/cjs/loader:813:14)
2021-06-05T05:47:34.659843+00:00 app[worker.1]:     at Module.require (node:internal/modules/cjs/loader:996:19)
2021-06-05T05:47:34.659843+00:00 app[worker.1]:     at require (node:internal/modules/cjs/helpers:92:18)
2021-06-05T05:47:34.666960+00:00 app[worker.1]: npm notice 
2021-06-05T05:47:34.667115+00:00 app[worker.1]: npm notice New minor version of npm available! 7.7.6 -> 7.16.0
2021-06-05T05:47:34.667195+00:00 app[worker.1]: npm notice Changelog: <https://github.com/npm/cli/releases/tag/v7.16.0>
2021-06-05T05:47:34.667295+00:00 app[worker.1]: npm notice Run `npm install -g npm@7.16.0` to update!
2021-06-05T05:47:34.667392+00:00 app[worker.1]: npm notice 
2021-06-05T05:47:34.728487+00:00 heroku[worker.1]: Process exited with status 1
2021-06-05T05:47:34.825795+00:00 heroku[worker.1]: State changed from up to crashed


Getting this error when I deploy 
Pls fix

**Reply by arghyac35** [reply]
You need to give a full log.. this isn't a full log.

**Comment by trickster-afk** [comment]
I've deployed the bot on Heroku following the instructions provided and the bot is up and running now.
When I try to mirror a torrent magnet link, the torrent gets downloaded and I face challenges while uploading it to my team drive.
Basically, there is an issue with the location of client_secret.json.

Errors captured in the Heroku logs:-

Error loading client secret file: ENOENT: no such file or directory, open './client_secret.json'
2021-06-06T05:14:42.292508+00:00 app[worker.1]: 6cd8f080a3c94ada: Failed to upload - /bot/downloads/1440502e-a251-4abd-8ee8-2b8b80248da0/All Dune books + short stories + extras ePUB: ENOENT: no such file or directory, open './client_secret.json'



Can someone please guide me to resolve this issue.
Thanks in advance.



![image](https://user-images.githubusercontent.com/85371057/120915178-fe27b380-c6bf-11eb-9381-3ac6d0f40fb7.png)


This is my client_secrets.json gist url. I have tried the name as client_secret.json as well as client_secrets.json.

This client_secret URL is being reflected properly in the heroku config vars.

**Reply by arghyac35** [reply]
Before the bot gets started, there should be logs like "Usage of token detected". I need the logs after this message.

**Reply by trickster-afk** [reply]
Thanks for the reply !! I have gone through the start script and resolved my mistake.
I actually configured the GIT related secrets, but later decided to not use SAs. So, the script was going into the first if condition and hence not fetching the client_secret.json file. 

I have now removed the GIT related secrets which were configured. Now everything is working fine.

But, I still have issues with configuring the SAs. I will give it another try, and if no success, I will post my issue here. Please do reply. 

Thanks.

**Comment by trickster-afk** [comment]
I'm facing issues with the SAs.

I have created an accounts folder and put all my SA JSONs in it. I have set all he GIT related secrets and the secrets are properly reflected in the Heroku's config vars.

The bot is up and running. Now when I mirror any link, it gets downloaded but fails while uploading it to my drive.

The error shown in the heroku logs is : -

2021-06-06T09:54:45.923257+00:00 app[worker.1]: Authorizing with nodemon.json service account
2021-06-06T09:54:45.924322+00:00 app[worker.1]: 3ea9f74a65cf686e: Failed to upload - /bot/downloads/89e495cf-acdb-4d6a-a401-3084e4e6dd27/All Dune books + short stories + extras ePUB: No key or keyFile set.


Any help here?

Thanks

**Reply by arghyac35** [reply]
Follow my SS of how service accounts should be uploaded to gh, also u need to maintain the folder name like that.

**Comment by tomyprs** [comment]
give an example to create a url button, for the gd link please 🥺

**Reply by BOI-Hub** [reply]
It gives u Index link for the mentioned Gdrive link. 
![Screenshot_20210608-005418~01](https://user-images.githubusercontent.com/71264215/121076694-5069f100-c7f4-11eb-852c-84d24261b214.jpg)

**Reply by arghyac35** [reply]
![IMG_20210628_074515_433.jpg](https://user-images.githubusercontent.com/27964664/123569889-cdc9c580-d7e4-11eb-9330-799a62fd8d65.jpg)

Add it like this way.

**Comment by rheido** [comment]
Can you implement zipping gdrive folders too besides tar?
/zip <driveurl> Create a zip of drive folder and upload.
I have seen some mirror bots which Supports gdrive folder zip and tar features but not opensource

**Reply by arghyac35** [reply]
No point in having same feature twice. You can use tar only. Motive is to download Google drive folder as compressed so be it as zip or tar it is same.

**Reply by rheido** [reply]
Oke sir i got it.

**Comment by askfriends** [comment]
bot got suspended by heroku, is this with everyone or just me?
